dataactvalidator/validation_handlers/validator.py

```diff
@@ -41,36 +47,28 @@ def cross_validate_sql(rules, submission_id, short_to_long_dict, job_id, error_c
             'status': 'start',
             'start': rule_start
         })
-        failed_rows = conn.execute(rule.rule_sql.format(submission_id))
-        logger.info({
-            'message': 'Finished running cross-file rule {} on submission_id: {}.'.format(rule.query_name,
-                                                                                          str(submission_id)) +
-                       'Starting flex field gathering and file writing',
-            'message_type': 'ValidatorInfo',
-            'rule': rule.query_name,
-            'job_id': job_id,
-            'submission_id': submission_id
-        })
-        if failed_rows.rowcount:
-            rule_cols = failed_rows.keys()
+
+        def process_failures(failures, columns, batch_num=0):
```

Comment:
If this is a separate function entirely, should we move it out of this function and just call it? The only reason all this processing was in here to begin with was because it was all being done inside this function. If we're making a new function we could move it out and make it easier to read.

Reply:
Agreeing with the sentiment though it turns out there are a bunch of shared scoped elements that would need to be converted to arguments. For example:
`def process_cross_failures(failures, columns, short_to_long_dict, rule, submission_id, job_id, error_csv, warning_csv, batch_num=0):`